src/nupic/engine/Region.hpp

```diff
@@ -461,50 +462,66 @@ class Region : public Serializable {
   void save_ar(Archive& ar) const {
     ar(cereal::make_nvp("name", name_),
        cereal::make_nvp("nodeType", type_),
+       cereal::make_nvp("initialized", initialized_),
        cereal::make_nvp("phases", phases_));
 
     std::map<std::string, Dimensions> outDims;
     std::map<std::string, Dimensions> inDims;
-    saveDims(outDims, inDims);
-    ar(cereal::make_nvp("outputs", outDims));
-    ar(cereal::make_nvp("inputs",  inDims));
+    getDims_(outDims, inDims);
+    ar(cereal::make_nvp("output_dims", outDims));
+    ar(cereal::make_nvp("input_dims",  inDims));
+		
+		// save the output buffers
+    std::map<std::string, Array> buffers;
+    getOutputBuffers_(buffers);
+	  ar(cereal::make_nvp("outputs", buffers));
+
     // Now serialize the RegionImpl plugin.
     ArWrapper arw(&ar);
     serializeImpl(arw);
   }
 
 
   // FOR Cereal Deserialization
+  // Note: custom region implementations must be registered
+  //       before deserializing a region.
```

Comment:
can we assert for that?

Reply:
>// Note: custom region implementations must be registered
  //       before deserializing a region.
 can we assert for that?

It is.  When it tries to hook it into the engine it will complain if it is not registered.